### ISSUE TITLE
prop-types doesn't check nextProps of componentWillReceiveProps

### ISSUE THREAD

**Issue by Eschon** [yannickcr/eslint-plugin-react#814]
This is kind of related to #801 but as far as I can tell the PR to fix (#792)  it doesn't fix my issue. I modified the reproduction case from #801 there to show my issue:

``` JavaScript
import React, {PropTypes} from "react";

export default React.createClass({

  propTypes: {
    bar: PropTypes.func,
  },

  componentWillReceiveProps (nextProps) {
    if (nextProps.foo) { // this line should trigger the 'prop-types' rule but doesn't
      return;
    }
  },

  render () {
    this.props.bar;
    return <div />;
  }
});
```

**Comment by mqklin** [comment]
@yannickcr seems like it isn't fixed: 
`react/prop-types` doesn't warn if I get props from `nextProps`
![image](https://user-images.githubusercontent.com/9456433/34660032-fa5473ae-f460-11e7-967c-5575f5f1d677.png)

  In this case there should be an Error about `x` as well.

**Comment by ljharb** [comment]
Since this is from a year ago, could you file a new issue for your use case?

**Comment by mqklin** [comment]
Yes, today I'll do it

**Comment by muhabalwan** [comment]
#1832 a related issue I'm guessing
